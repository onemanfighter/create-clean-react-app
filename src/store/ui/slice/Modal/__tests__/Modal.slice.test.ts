import { ModalID, appStore } from '@uiStore';
import { act, renderHook } from '@testing-library/react-hooks';

describe('Modal slice', () => {
  const onModalCloseMock = jest.fn();

  it('should return modal selector state and actions', () => {
    const modal = renderHook(() => appStore()).result.current;

    expect(modal.Modal).toMatchSnapshot();
  });

  it('should return modal state with initial value', () => {
    const { result } = renderHook(() => appStore());

    expect(result.current.Modal).toMatchSnapshot();
  });

  it('should return modal state on modal data set using openModal', () => {
    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Modal.openModal(ModalID.SEARCH, onModalCloseMock);
      jest.runAllTimers();
    });

    expect(result.current.Modal).toMatchSnapshot();
  });

  it('should reset and close modal state on modal data set using closeModal', () => {
    const { result } = renderHook(() => appStore());

    result.current.Modal.openModal(ModalID.SEARCH, onModalCloseMock);

    expect(result.current.Modal.modalID).toEqual(ModalID.SEARCH);
    expect(result.current.Modal.modalData?.onModalClose).not.toBeUndefined();

    result.current.Modal.resetModalState();
    jest.runAllTimers();

    expect(result.current.Modal.modalID).toEqual(ModalID.NONE);
    expect(result.current.Modal.modalData?.onModalClose).toBeUndefined();
  });
});
