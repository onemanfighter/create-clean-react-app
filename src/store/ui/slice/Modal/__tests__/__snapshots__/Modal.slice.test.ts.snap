// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal slice should return modal selector state and actions 1`] = `
{
  "modalData": undefined,
  "modalID": "none",
  "modalOpenState": "CLOSE",
  "openModal": [Function],
  "resetModalState": [Function],
}
`;

exports[`Modal slice should return modal state on modal data set using openModal 1`] = `
{
  "modalData": {
    "onModalClose": [MockFunction],
  },
  "modalID": "search",
  "modalOpenState": "OPEN",
  "openModal": [Function],
  "resetModalState": [Function],
}
`;

exports[`Modal slice should return modal state with initial value 1`] = `
{
  "modalData": undefined,
  "modalID": "none",
  "modalOpenState": "CLOSE",
  "openModal": [Function],
  "resetModalState": [Function],
}
`;
